hoomd/md/pytest/test_rigid.py

```diff
@@ -453,3 +453,152 @@ def test_rigid_dof(
         thermo_central_free.translational_degrees_of_freedom == (n_bodies + n_free) * 3
     )
     assert thermo_constituent.translational_degrees_of_freedom == 0
+
+
+@pytest.mark.parametrize(
+    "com_velocity",
+    [(0, 0, 0), (1, 2, 3)],
+)
+def test_velocity_constituents(
+    com_velocity,
+    simulation_factory,
+    one_particle_snapshot_factory,
+    valid_body_definition,
+):
+    """Test velocity of constituents updates correctly with constant torque."""
+    initial_snapshot = one_particle_snapshot_factory(particle_types=["A", "B"])
+    if initial_snapshot.communicator.rank == 0:
+        initial_snapshot.particles.moment_inertia[:] = [(1, 1, 1)]
+        initial_snapshot.particles.velocity[:] = [com_velocity]
+
+    sim = simulation_factory(initial_snapshot)
+    sim.seed = 5
+
+    rigid = md.constrain.Rigid()
+    rigid.body["A"] = valid_body_definition
+    rigid.create_bodies(sim.state)
+
+    constvol = md.methods.ConstantVolume(filter=hoomd.filter.Rigid())
+    integrator = md.Integrator(
+        dt=0.005,
+        methods=[constvol],
+        integrate_rotational_dof=True,
+    )
+    integrator.rigid = rigid
+    sim.operations += integrator
+    sim.run(1)
+
+    new_snapshot = sim.state.get_snapshot()
+    expected_velocity = [com_velocity] * 5
+    assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
```

Comment:
For compatibility with MPI, you need to only access snapshots on the root rank (0):
```suggestion
    if new_snapshot.communicator.rank == 0:
        expected_velocity = [com_velocity] * 5
        assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
```